### PR TITLE
show created users in sys.users table

### PR DESCRIPTION
**Pull request by bogensberger**

### CODE REVIEW

File: users/src/main/java/io/crate/operation/user/UserManagerService.java [hunk 1]
```diff
@@ -18,20 +18,26 @@
...
 import java.util.concurrent.CompletableFuture;
 
 import static io.crate.operation.user.UsersMetaData.PROTO;
 import static io.crate.operation.user.UsersMetaData.TYPE;
 
-public class UserManagerService implements UserManager {
+public class UserManagerService implements UserManager, ClusterStateListener {
 
-    private final User crateUser = new User("crate", true);
+    private static User crateUser = new User("crate", true);
```
Comment (chaudum) [inline users/src/main/java/io/crate/operation/user/UserManagerService.java]: static filed should be uppercase, also, can be final

File: users/src/main/java/io/crate/operation/user/UserManagerService.java [hunk 2]
```diff
@@ -61,6 +69,24 @@
...
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (!event.metaDataChanged()) {
+            return;
+        }
+        users = getNewCurrentUsers(event.state().metaData().custom(UsersMetaData.TYPE));
+    }
+
+    static Set<User> getNewCurrentUsers(@Nullable UsersMetaData metaData) {
```
Comment (chaudum) [inline users/src/main/java/io/crate/operation/user/UserManagerService.java]: this method name is a bit weird. current users, or new users?
I think something like `getUsersFromClusterState()` would be a better name

File: users/src/test/java/io.crate.operation.user/UserManagerServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import org.junit.Test;
+
+import java.util.Set;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+public class UserManagerServiceTest extends CrateUnitTest {
+
+    private User crateUser = new User("crate", true);
```
Comment (chaudum) [inline users/src/test/java/io.crate.operation.user/UserManagerServiceTest.java]: you could use the `crate` user from `UserManagerService` if you made it package local

File: users/src/main/java/io/crate/operation/user/UserManagerService.java [hunk 4]
```diff
@@ -18,20 +18,26 @@
...
 
 import static io.crate.operation.user.UsersMetaData.PROTO;
 import static io.crate.operation.user.UsersMetaData.TYPE;
 
-public class UserManagerService implements UserManager {
+public class UserManagerService implements UserManager, ClusterStateListener {
 
-    private final User crateUser = new User("crate", true);
+    static User CRATE_USER = new User("crate", true);
+    private Set<User> users = ImmutableSet.of(CRATE_USER);
```
Comment (mfussenegger) [inline users/src/main/java/io/crate/operation/user/UserManagerService.java]: shouldn't this be volatile ?

Comment (bogensberger) [inline users/src/main/java/io/crate/operation/user/UserManagerService.java]: you are right, updated it

**Pull request merged**